### PR TITLE
Implement EAR Deployment Descriptors for JAVA EE / Jakarta EE

### LINKED ISSUE #26438: Ear Plugin: XML Schema / xmlns in Ear Descriptor for Java EE 8, Jakarta EE 9 and Jakarta EE 10 is missing / invalid
### Issue type

Polishing

### Problem description

I would expect the `version` Property for `ear.deploymentDescriptor` to fully support values up to `"10"` (Jakarta EE 10).

Currently the documentation of the `version` property (comment / javadoc in DeploymentDescriptor.java) states it only supports values up to "6" while in fact it currently already properly handles "7".
Higher values lead to a descriptor with invalid/missing xmlns.

A desciptor like this:
```gradle
ear {
    deploymentDescriptor {
        version = "10" // <--- it currently only fully supports values up to "7"
        applicationName = "example"
        displayName = "Just an Example"
        // ... remaining properties omitted here ...
    }
}
```
Should generate the descriptor like this:
_(With valid xmlns etc. according to the JAVA EE / Jakarta EE Schemas)_
```xml
<?xml version="1.0"?>
<application xmlns="https://jakarta.ee/xml/ns/jakartaee" xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/application_10.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="10">
  <application-name>example</application-name>
  <display-name>Just an Example</display-name>
  ... remaining elements omitted here ...
</application>
```

**However currently the application Element looks like this:**
```xml
<?xml version="1.0"?>
<application xmlns="" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="10">
  <application-name>example</application-name>
  <display-name>Just an Example</display-name>
  ... remaining elements omitted here ...
</application>
```

- **xmlns is an empty string** for Versions 8, 9 and 10
- **xsi:schemaLocation is missing** for Versions 8, 9 and 10

### Context (optional)

The latest XSD Schemas can be found at https://jakarta.ee/xml/ns/jakartaee/

https://www.oracle.com/webfolder/technetwork/jsc/xml/ns/javaee/application_8.xsd (JAVA EE 8)
```xml
<application xmlns="http://xmlns.jcp.org/xml/ns/javaee"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee 
      	http://xmlns.jcp.org/xml/ns/javaee/application_8.xsd"
      version="8">
      ...
</application>
```
---
https://jakarta.ee/xml/ns/jakartaee/application_9.xsd (Jakarta EE 9)
```xml
<application xmlns="https://jakarta.ee/xml/ns/jakartaee"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee 
      	https://jakarta.ee/xml/ns/jakartaee/application_9.xsd"
      version="9">
      ...
</application>
```
---
https://jakarta.ee/xml/ns/jakartaee/application_10.xsd (Jakarta EE 10)
```xml
<application xmlns="https://jakarta.ee/xml/ns/jakartaee"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee 
      	https://jakarta.ee/xml/ns/jakartaee/application_10.xsd"
      version="10">
      ...
</application>
```

### PR DESCRIPTION
**Pull request by p-schneider**
Support JAVA EE 8, Jakarta EE 9 and Jakarta EE 10 in ear Descriptor by correctly handling Values "8", "9", "10" in property `ear.deploymentDescriptor.version`.

### Context

See #26438

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by p-schneider** [comment]
This PR only adds the correct xmlns / schema properties to for Versions 8, 9 and 10.
The default value is still 6, that wasn't changed in this PR.

It could however make sense to also update the default value of the version property to a recent version, e.g. "10" (=Jakarta EE 10) in the near future.

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -50,6 +51,19 @@ Example:
 ADD RELEASE FEATURES BELOW
 vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->
 
+<a name="ear-plugin"></a>
+### Ear Plugin
+
+It is now possible to generate valid deployment descriptors for Java EE 8, Jakarta EE 9 and Jakarta EE 10
+by specifying the corresponding version in the `deploymentDescriptor` instead of having to use a custom descriptor file.
+
+```groovy
```
Comment (eskatos) [inline subprojects/docs/src/docs/release/notes.md:60]: ❌ We use Kotlin snippets by default in release notes

**Review by eskatos — APPROVED** [review]
Only one comment about the release notes

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (cobexer) [reply subprojects/docs/src/docs/release/notes.md]: updated

**Comment by cobexer** [comment]
@bot-gradle test and merge

**Review by lkasso — APPROVED** [review]

**Pull request merged**